src/main/java/amazin/model/User.java

```diff
@@ -0,0 +1,102 @@
+package amazin.model;
+
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import java.util.Set;
+
+@Entity
+@Indexed
+@Table(name="users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "user_id", updatable = false, nullable = false)
+    private Long id;
+    @NotNull
+    @Size(min = 1, message = "First name cannot be empty")
+    @Column(name="user_first_name")
+    @Field
+    private String firstName;
+    @NotNull
+    @Size(min = 1, message = "Last name cannot be empty")
+    @Column(name="user_last_name")
+    @Field
+    private String lastName;
+    @NotNull
+    @Pattern(regexp = "^\\w+@[a-zA-Z_]+?\\.[a-zA-Z]{2,3}$", message = "Not a valid email address")
+    @Field
+    @Column(unique = true, name="user_email")
+    private String email;
+    @NotNull
+    @Size(min = 8, message = "Password must be at least 8 characters")
```

Comment:
you could use @min() and @max() annotations

Reply:
This is just a syntax change not a semantic change. Aaron needed to use one and just stay consistent, and he was consistent